### ISSUE TITLE
JSX parser not working with style-it

### ISSUE THREAD

**Issue by ronak3110** [TroyAlford/react-jsx-parser#132]
Hi, we are trying to add css string to a react component which is parsed from a jsx string.

For the css we are using Style-it and for the jsx parsing we are using react-jsx-parser.

in my render function,

Style.it(`
      .intro {
        font-size: 40px;
      }
    `,
      <JsxParser
    bindings={{
      myEventHandler: () => { /* ... do stuff ... */ },
    }}
    components={InjectableComponent}
    jsx={`
      <h1 className='intro '>Header</h1>
      <InjectableComponent eventHandler={myEventHandler} truthyProp />
    `}
  />
    );

the parsed jsx is rendered but with no styles, 
the style-it is working if i provide any other react component or a valid jsx.
when jsx parser is used its not working.

**Comment by TroyAlford** [comment]
Yes, this makes sense. If you look at the in/out examples on the `style-it` website, it shows this pattern:
```
Style.it(`.intro { font-size: 40px; }`, <p className="intro">Foo</p>)
```
The output is then:
```
<p class="intro _scoped_1">
  <style type="text/css">
    ._scoped-1.intro { font-size: 40px; }
  </style>
  Foo
</p>
```

In your case, you are going to wind up with an output of:
```
  <div class="jsx-parser">
    <style type="text/css">._scoped-1.intro { font-size: 40px; }</style>
    <h1 class="intro">...</h1>
  </div>
```

You would expect the wrapper div to have `class="_scoped-1 jsx-parser"` - but it doesn't, because the `JsxParser` component doesn't currently support the `className` prop. As a result, the `_scoped-1` class isn't added _anywhere_ - so the selectors don't work. It would be easy enough to make this lib support the `className` property - which would resolve this handily, but might be confusing if someone used `renderInWrapper={false}` - because then the `className` would be thrown away. And - there's an easy workaround for what you're after - just disable the default wrapper `div`, and supply your own:

```
Style.it(`.intro { font-size: 40px; }`,
  <div className="intro">
    <JsxParser jsx="<h1 className='intro'>Foo</h1>" renderInWrapper={false} />
  </div>
)
```

This will result in the following:
```
<div class="_styled-1 intro">
  <style type="text/css">._styled-1.intro { font-size: 40px; }</style>
  <h1 class="intro">Foo</h1>
</div>
```

You don't technically _need_ the `renderInWrapper={false}` - but it's probably superfluous in your case to render the extra wrapper.

**Comment by TroyAlford** [comment]
FYI - I tested this in codesandbox - and the workaround works as-expected.

**Comment by TroyAlford** [comment]
This is now fixed & released as [v1.25.0](https://github.com/TroyAlford/react-jsx-parser/releases/tag/v1.25.0)

**Comment by ronak3110** [comment]
Thanks, with v1.25.0 its working as expected.
